bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -266,9 +267,18 @@ private void decodeProperties(final BsonReader reader, final DecoderContext deco
         return codec;
     }
 
+    @SuppressWarnings("unchecked")
     private <S> Codec<S> getCodecFromPropertyRegistry(final PropertyModel<S> propertyModel) {
         try {
-            return propertyCodecRegistry.get(propertyModel.getTypeData());
+            Codec<S> codec = propertyCodecRegistry.get(propertyModel.getTypeData());
+            BsonType representation = propertyModel.getBsonRepresentation();
+            if (representation != null) {
+                if (codec instanceof RepresentationConfigurable) {
+                    return ((RepresentationConfigurable<S>) codec).withRepresentation(representation);
```

Comment:
I'm not clear why this call to withRepresentation couldn't be done earlier, like in PropertyModelBuilder#build, instead of delayed until this point.

Reply:
This is actually the first time we get the `codec` we need as far as I can tell (line 273, the call to propertyCodecRegistry#get is where we get it). The codec is always null in PropertyModelBuilder#build as far as I can tell. I believe this is the intended behavior, as we only get the codec when we need it the first time, and then it is cached.

Reply:
OK.